wandb/sdk/internal/internal_api.py

```diff
@@ -254,10 +254,9 @@ def __init__(
 
         # todo: remove this hacky hack after settings refactor is complete
         #  keeping this code here to limit scope and so that it is easy to remove later
-        extra_http_headers = self.settings(
-            "_extra_http_headers"
-        ) or wandb.sdk.wandb_settings._str_as_json(
-            self._environ.get("WANDB__EXTRA_HTTP_HEADERS", {})
+        extra_http = self._environ.get("WANDB__EXTRA_HTTP_HEADERS", "{}")
+        extra_http_headers = self.settings("_extra_http_headers") or json.loads(
+            extra_http
         )
```

Comment:
Nit: `extra_http` is not a great variable name...
```suggestion
        extra_http_headers = self.settings("_extra_http_headers") or json.loads(
            self._environ.get("WANDB__EXTRA_HTTP_HEADERS", "{}")
        )
```